### PR TITLE
Rename + integrate resource cluster routes and actor.

### PR DESCRIPTION
**Pull request by Andyz26**
### Context
[Update: Due to some broken tests this PR is updated to include the integration of host actor into main entry point + config etc.]

Rename routes used by resource cluster actors to be one redirected and one rejected if not on leader node.
Merge host route into redirect route + remove TestNG tests.

### Checklist

- [ ] `./gradlew build` compiles code correctly
- [ ] Added new tests where applicable
- [ ] `./gradlew test` passes all tests
- [ ] Extended README or added javadocs where applicable
- [ ] Added copyright headers for new files from `CONTRIBUTING.md`

### CODE REVIEW

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersRejectNonLeaderRoute.java [hunk 1]
```diff
@@ -41,7 +41,7 @@
  */
 @Slf4j
 @RequiredArgsConstructor
-public class ResourceClustersWriteRoute extends BaseRoute {
+public class ResourceClustersRejectNonLeaderRoute extends BaseRoute {
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersRejectNonLeaderRoute.java:44]: nit: maybe rename it to ResourceClustersLeaderExclusiveRoute

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+ * /api/v1/resourceClusters/{}/getResourceOverview                    (GET)
+ * /api/v1/resourceClusters/{}/getRegisteredTaskExecutors             (GET)
+ * /api/v1/resourceClusters/{}/getBusyTaskExecutors                   (GET)
+ * /api/v1/resourceClusters/{}/getAvailableTaskExecutors              (GET)
+ * /api/v1/resourceClusters/{}/getUnregisteredTaskExecutors           (GET)
+ * <p>
+ * /api/v1/resourceClusters/{}/taskExecutors/{}/getTaskExecutorState  (GET)
+ */
+@Slf4j
+//@RequiredArgsConstructor
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java:65]: nit: remove this if it's not required.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+        segment("api").slash("v1").slash("resourceClusters");
+
+    private final ResourceClusters gateway;
+
+    private final ResourceClusterRouteHandler resourceClusterRouteHandler;
+    private final Cache<Uri, RouteResult> routeResultCache;
+
+    // TODO(andyz) integrate w/ masterMain
+    @VisibleForTesting
+    public ResourceClustersNonLeaderRedirectRoute(ResourceClusters gateway) {
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java:78]: make this package-private if it's only for testing.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+ * Defines the following end points:
+ * /api/v1/resourceClusters/list                                      (GET)
+ * <p>
+ * /api/v1/resourceClusters/{}/getResourceOverview                    (GET)
+ * /api/v1/resourceClusters/{}/getRegisteredTaskExecutors             (GET)
+ * /api/v1/resourceClusters/{}/getBusyTaskExecutors                   (GET)
+ * /api/v1/resourceClusters/{}/getAvailableTaskExecutors              (GET)
+ * /api/v1/resourceClusters/{}/getUnregisteredTaskExecutors           (GET)
+ * <p>
+ * /api/v1/resourceClusters/{}/taskExecutors/{}/getTaskExecutorState  (GET)
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java:62]: Documentation: Add the other routes here.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+//@RequiredArgsConstructor
+public class ResourceClustersNonLeaderRedirectRoute extends BaseRoute {
+
+    private static final PathMatcher0 RESOURCECLUSTERS_API_PREFIX =
+        segment("api").slash("v1").slash("resourceClusters");
+
+    private final ResourceClusters gateway;
+
+    private final ResourceClusterRouteHandler resourceClusterRouteHandler;
+    private final Cache<Uri, RouteResult> routeResultCache;
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java:74]: what's the function of this?

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/proto/ResourceClusterAPIProto.java [hunk 6]
```diff
@@ -17,25 +17,32 @@
...
 import java.util.List;
 import lombok.Builder;
 import lombok.Singular;
 import lombok.Value;
 
 public class ResourceClusterAPIProto {
 
+    // TODO: @Value generated equals doesn't include base class fields.
     @Value
     public static class ListResourceClustersResponse extends BaseResponse {
```
Comment (sundargates) [inline mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/proto/ResourceClusterAPIProto.java:31]: are there tests to verify the structure of these Jackson serialized data structures? If not, can we add them?

**Review by sundargates — APPROVED** [review]

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/proto/ResourceClusterAPIProto.java (hunk 6 above)
Comment (Andyz26) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/proto/ResourceClusterAPIProto.java:31]: Added another TODO in one of the tests that covers this.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java (hunk 5 above)
Comment (Andyz26) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java:77]: It's caching the result of spec file. Nothing critical I think but nice to have with a small TTL.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java (hunk 3 above)
Comment (Andyz26) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java]: Removed.

File: mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java (hunk 5 above)
Comment (codyrioux) [reply mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/v1/ResourceClustersNonLeaderRedirectRoute.java:77]: I agree, a very small TTL here on a cache bounds the incoming load to the master and buys us some more scalability.

**Review by codyrioux — APPROVED** [review]

**Pull request merged**